### ISSUE TITLE
`UndefinedMagicMethod` on third-party facades — FacadeMapProvider only scans Testbench's AliasLoader

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#761]
## Description

Any facade that is only registered by the **analysed project's** service providers (as opposed to the plugin's internal Orchestra Testbench app) is invisible to the plugin. Calls on such a facade emit `UndefinedMagicMethod`, because Psalm sees a class whose `@method` catalogue is empty and whose `__callStatic` has no known targets.

This is structurally distinct from #591 (plugin *handlers* not firing on proxied calls). #591 assumes the call is *valid* and asks the plugin to propagate its handlers through `__callStatic`. This issue is one step earlier: the call is flagged as invalid in the first place, because the plugin has no record that the facade exists at all.

## Reproduction

Audited [spatie/laravel-backup](https://github.com/spatie/laravel-backup) with the plugin (v4.8.2, Psalm 7 beta, `errorLevel=1`). The `Signal` facade (`Spatie\SignalAwareCommand\Facades\Signal`) is registered by `spatie/laravel-signal-aware-command`'s `SignalAwareCommandServiceProvider`, which is not loaded by the plugin's Testbench app:

```
ERROR: UndefinedMagicMethod - src/Tasks/Backup/BackupJob.php:178
  Magic method Spatie\SignalAwareCommand\Facades\Signal::handle does not exist

ERROR: UndefinedMagicMethod - src/Tasks/Backup/BackupJob.php:210
  Magic method Spatie\SignalAwareCommand\Facades\Signal::clearhandlers does not exist
```

Corresponding code:

```php
// src/Tasks/Backup/BackupJob.php
Signal::handle(SIGINT, function (Command $command) {
    $command->info('Cleaning up temporary files...');
});
// ...
Signal::clearHandlers(SIGINT);
```

`Signal` extends `Illuminate\Support\Facades\Facade`, defines `getFacadeAccessor()`, and has the usual dispatcher class behind it — but its alias is never registered in `AliasLoader::getInstance()->getAliases()` during the plugin's boot, so the plugin never indexes it.

The same class of false positive fires on every third-party facade: Livewire, Filament, Horizon, Nova, Spatie's broader stable, and any first-party facade an application registers under `config/app.php` `aliases`. Larastan handles this by statically discovering `Facade` subclasses from the analysed codebase itself, independent of any boot.

## Root cause

[`FacadeMapProvider::init()`](https://github.com/psalm/psalm-plugin-laravel/blob/master/src/Providers/FacadeMapProvider.php#L33-L98) reads only the booted Testbench app's alias registry:

```php
$aliases = AliasLoader::getInstance()->getAliases();

foreach ($aliases as $alias => $facadeClass) {
    // ...
    if (!is_subclass_of($facadeClass, Facade::class)) { continue; }
    $root = $facadeClass::getFacadeRoot();
    // ...
}
```

That registry contains only Laravel-core facades (`Auth`, `Cache`, `DB`, `View`, etc.) because the plugin's `ApplicationProvider` boots a vanilla Testbench app and does not load the analysed project's service providers. Third-party facade classes are therefore never mapped, even though they are physically present under `vendor/` and reachable by the autoloader.

Beyond `FacadeMapProvider`, no other part of the plugin tries to surface third-party facades to Psalm either — the plugin has no generic "discover every `Facade` subclass in scope" pass. As a result:

- `FacadeMapProvider::getFacadeClasses($serviceClass)` returns an empty list for any third-party service, so plugin handlers never register against the facade.
- No generated alias stub, no `@method` catalogue, nothing for Psalm to resolve `Signal::handle()` against.
- Psalm therefore falls back to `__callStatic` magic resolution — and emits `UndefinedMagicMethod`.

## Proposed approaches

None of these is a trivial fix; listing the options with tradeoffs so the maintainer can pick the scope:

### Option A (minimal) — static Facade discovery during scan

During `AfterClassLikeVisit`, for every class that extends `Illuminate\Support\Facades\Facade`:

1. Statically resolve the facade's `getFacadeAccessor()` — read the return value by AST if it is a simple string literal, or fall back to reflection against the autoloader.
2. Resolve that accessor to a concrete class. For literal class strings (`SomeService::class`) this is trivial; for container aliases (`'cache'`, etc.) the plugin's booted app can still be asked.
3. Register the facade in `FacadeMapProvider::$serviceToFacades`. Plugin handlers now fire for third-party facades.

This pass is independent of the plugin's Testbench boot and would also cover facades registered under `config/app.php` `aliases` in regular applications.

### Option B (fuller) — generate alias stubs for every `Facade` subclass

Emit a synthetic stub per discovered facade with a `@method` catalogue derived from the accessor's public methods. This fixes `UndefinedMagicMethod` at the Psalm level even when handlers aren't relevant. More expensive, but covers the most common user-visible symptom directly.

### Option C (pragmatic) — fall back when a facade is seen

Narrower scope: when Psalm asks about a method on a known `Facade` subclass that isn't in `FacadeMapProvider`, return a `mixed` resolution rather than letting Psalm emit `UndefinedMagicMethod`. This is a regression-in-type-precision in exchange for removing the false positive, and is closest in spirit to how Larastan's `ReturnTypes\FacadeExtension` degrades.

Option A is probably the cleanest first step and does not preclude B/C later.

## Impact

Every analysed package or application that uses a facade defined outside `illuminate/support` surfaces at least one `UndefinedMagicMethod` per call site. Small count per package (2x in `spatie/laravel-backup`) but universal across the ecosystem:

- Livewire, Filament, Horizon, Nova, Octane, Pulse, Pennant, Cashier.
- All Spatie packages that ship a facade (Permission, Medialibrary, Signal, etc.).
- Any first-party application facade registered via `config/app.php` `aliases` or `extra.laravel.aliases` in a package's `composer.json`.

Related but distinct: #591 (plugin handlers not firing on valid proxied calls). Solving this issue unblocks progress on #591 for third-party facades — you cannot forward handlers through a `__callStatic` that the plugin doesn't know about.

**Comment by alies-dev** [comment]
Subset of #766 (project-provider blindness during package analysis). Same root cause: when Testbench is the fallback in `ApplicationProvider::doGetApp()` (i.e. analysing a Laravel **package**, not a Laravel app), only the analysed package's providers are booted, so `FacadeMapProvider` cannot see facades registered by sibling packages.

Tracking #766 as the umbrella; this issue captures the third-party facade `__callStatic` symptom (`UndefinedMagicMethod`).

**Comment by alies-dev** [comment]
## Rescoped

The general root cause described above ("the plugin has no generic *discover every `Facade` subclass* pass; `FacadeMapProvider` only reads Testbench's `AliasLoader`") is now **outdated**. `AppFacadeRegistrationHandler` (shipped for #787) implements the **Option A** static-discovery pass proposed here: during `AfterClassLikeVisit` it finds every concrete `Facade` subclass in Psalm's scanned storage, probes `getFacadeRoot()`, queues the root for scanning, and binds `FacadeMethodHandler` (existence + params + return type) per facade FQCN. Only `Illuminate\`, `Laravel\`, `Mockery\`, and `PHPUnit\` are skipped — third-party namespaces (Spatie, Livewire, Filament, …) are discovered. The umbrella #766 was closed for the same reason.

So the original symptom is fixed for the common cases. What remains is one narrow residual, which this issue now tracks:

### Scope (residual only)

`UndefinedMagicMethod` on a third-party facade whose `getFacadeAccessor()` returns a **string container-alias** (e.g. `'signal'`) that is bound **only by a sibling package's service provider**, **and only when the plugin is analysing a Laravel package** (the Testbench fallback in `ApplicationProvider::doGetApp()` boots only the analysed package's own providers, so the sibling alias is unbound). `getFacadeRoot()` then throws, `AppFacadeRegistrationHandler::tryGetFacadeRootClass()` returns null, no provider is registered, and the call is flagged.

### Out of scope (already resolved / by design)

- **Facades with a class-string accessor** (`getFacadeAccessor()` returns `SomeService::class`): resolved — the container auto-wires them via reflection regardless of provider boot.
- **Application analysis** (not package analysis): resolved — the plugin boots the user's real kernel, so all providers, bindings, and facades are visible.
- **`Laravel\` first-party sub-packages** (Cashier, Horizon, Telescope, Pulse, Octane, Pennant): intentionally skipped — probing their accessors chains into `BindingResolutionException`. Tracked separately if needed.

### Likely fix direction

Same as the closed #766 root cause: surface the analysed package's own service providers to the Testbench boot (e.g. read `extra.laravel.providers` from the package's `composer.json`) so sibling/own string-alias bindings resolve. Until then this is a known package-analysis limitation.

**Comment by alies-dev** [comment]
I think we can close this one
